android-json-form-wizard/src/main/java/com/vijay/jsonwizard/interactors/JsonFormInteractor.java

```diff
@@ -173,16 +181,42 @@ public void fetchFields(List<View> viewsFromJson, String stepName, JsonFormFragm
 
             for (int i = 0; i < fields.length(); i++) {
                 JSONObject childJson = fields.getJSONObject(i);
-                fetchViews(viewsFromJson, stepName, formFragment, childJson.getString(JsonFormConstants.TYPE), childJson,
+                List<View> views = fetchViews(viewsFromJson, stepName, formFragment, childJson.getString(JsonFormConstants.TYPE), childJson,
                         listener, popup);
+                if (childJson.get(TYPE).equals(REPEATING_GROUP)) {
+                    for (int j = i + 1; j < fields.length(); j++) {
+                        JSONObject repeatingChildJson = fields.getJSONObject(j);
+                        if (repeatingChildJson.has(PARENT_REPEATING_GROUP) && repeatingChildJson.getString(PARENT_REPEATING_GROUP).equals(childJson.getString(KEY))) {
+                            List<View> childViews = fetchViews(viewsFromJson, stepName, formFragment, repeatingChildJson.getString(JsonFormConstants.TYPE), repeatingChildJson,
+                                    listener, popup);
+
+                            final int childCount = childViews.size();
+                            for (int index = 0; index < childCount; index++) {
+                                View v = childViews.get(index);
+                                if (views.size() > 0 ) {
+                                    if (v.getParent() != null) {
+                                        ((ViewGroup) v.getParent()).removeView(v);
+                                    }
+                                    viewsFromJson.removeAll(childViews);
+                                    String[] uniqueID = repeatingChildJson.getString(KEY).split("_");
+                                    v.setTag(R.id.repeating_group_key, uniqueID[uniqueID.length - 1]);
+                                    ((ViewGroup) views.get(0)).addView(v);
+                                }
+                            }
+                            i = j;
+                        } else {
+                            break;
+                        }
+                    }
+                }
```

Comment:
Could you please give more information on what is being done here?

Reply:
Adding Repeating group child section just after repeating group So that when you come back to the repeated group section step, all children should be visible after that Repeating group.
FIXED: Should add repeating group sections after repeating the group field, not at the end step.